### PR TITLE
Updating the explanation for Network type

### PR DESCRIPTION
**Pull request by alexanderConstantinescu**
This PR addresses the bug report [1729311](https://bugzilla.redhat.com/show_bug.cgi?id=1729311)

This PR concerns only the documentation of the network resource.  

Please have a look at the wording as I am not sure that's what should be mentioned (@squeed )

/assign @squeed

### CODE REVIEW

**Comment by alexanderConstantinescu** [comment]
/assign @danwinship 

@danwinship Could you please help me with the paraphrasing this correctly?

**Comment by alexanderConstantinescu** [comment]
> Hm... it's odd that `oc explain` shows the doc comment for the `Network` object's `Spec` field, but doesn't show the documentation for the `NetworkSpec` struct

Come to think of it, that's the whole issue with this. The doc comment for NetworkSpec mentions the usage and "Gotchas":

> // NetworkSpec is the desired network configuration.
// As a general rule, this SHOULD NOT be read directly. Instead, you should
// consume the NetworkStatus, as it indicates the currently deployed configuration.
// Currently, changing ClusterNetwork, ServiceNetwork, or NetworkType after
// installation is not supported.

I will remove my added comments and try to find out and fix whatever bug there might be with the print of this comment.

**Comment by alexanderConstantinescu** [comment]
Couldn't get to the bottom of what prevents the docstring above a sub-struct to get exposed to `oc explain` so I filed [an issue instead](https://github.com/openshift/api/issues/407) and updated this PR with a work-around for the current issue.

File: config/v1/types_network.go [hunk 1]
```diff
@@ -6,27 +6,28 @@ import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
...
 	metav1.TypeMeta `json:",inline"`
 	// Standard object's metadata.
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
 	// spec holds user settable values for configuration.
+	// NetworkSpec is the desired network configuration.
+	// As a general rule, this SHOULD NOT be read directly. Instead, you should
+	// consume the NetworkStatus, as it indicates the currently deployed configuration.
+	// Currently, changing ClusterNetwork, ServiceNetwork, or NetworkType after
+	// installation is not supported.
```
Comment (danwinship) [inline config/v1/types_network.go]: Drop the "`NetworkSpec is the desired network configuration.`" line

Having thought about this some more, I don't think we should be listing the immutable fields here. People will forget to change the comment here when they add new fields in the struct below, etc. Instead, maybe say something about "most" of the fields being immutable after installation here, and clarify the specifics on the individual fields in `NetworkSpec`.

File: config/v1/types_network.go [hunk 2]
```diff
@@ -6,27 +6,28 @@ import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
...
 	// +kubebuilder:validation:Required
 	// +required
 	Spec NetworkSpec `json:"spec"`
 	// status holds observed values from the cluster. They may not be overridden.
+	// NetworkStatus is the current network configuration.
```
Comment (danwinship) [inline config/v1/types_network.go]: We don't need the addition here; the existing doc comment is enough

File: config/v1/types_network.go [hunk 3]
```diff
@@ -6,27 +6,28 @@ import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
...
 	// +optional
 	Status NetworkStatus `json:"status"`
 }
 
-// NetworkSpec is the desired network configuration.
-// As a general rule, this SHOULD NOT be read directly. Instead, you should
-// consume the NetworkStatus, as it indicates the currently deployed configuration.
-// Currently, changing ClusterNetwork, ServiceNetwork, or NetworkType after
-// installation is not supported.
```
Comment (danwinship) [inline config/v1/types_network.go]: Leave the doc comments on `NetworkSpec` and `NetworkStatus` (but remove the list of immutable fields as above). Even though they don't show up in `oc describe`, they still show up in other places.

File: config/v1/types_network.go [hunk 4]
```diff
@@ -6,27 +6,28 @@ import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
...
 type NetworkSpec struct {
 	// IP address pool to use for pod IPs.
 	ClusterNetwork []ClusterNetworkEntry `json:"clusterNetwork"`
```
Comment (danwinship) [inline config/v1/types_network.go]: so, eg, we should add "Currently this cannot be changed after installation" to this comment, etc

File: config/v1/types_network.go [hunk 5]
```diff
@@ -6,14 +6,17 @@ import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 // +genclient:nonNamespaced
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
-// Network holds cluster-wide information about Network.  The canonical name is `cluster`
-// TODO this object is an example of a possible grouping and is subject to change or removal
+// Network holds cluster-wide information about Network. The canonical name is `cluster`. It is used to configure the desired network configuration, such as: IP address pools for services/pod IPs, network plugin, etc. 
+// Please view network.spec for an explanation on what is applies when configuring this resource.
```
Comment (rcarrillocruz) [inline config/v1/types_network.go]: s/applies/applied

Comment (alexanderConstantinescu) [reply config/v1/types_network.go]: Indeed, thanks!

**Pull request merged**